### PR TITLE
Fix defcustom type specification for pairs-alist variable

### PR DESCRIPTION
**Pull request by wbolster**
this fixes the `:type` specification for `evil-surround-pairs-alist` so that the customize interface can do something sensible with it. (partial) screenshot after applying this fix:

image

### CODE REVIEW

File: evil-surround.el [hunk 1]
```diff
@@ -65,8 +65,11 @@ Each item is of the form (TRIGGER . (LEFT . RIGHT)), all strings.
 Alternatively, a function can be put in place of (LEFT . RIGHT).
 This only affects inserting pairs, not deleting or changing them."
   :group 'surround
-  :type '(repeat (cons (regexp :tag "Key")
-                       (symbol :tag "Surround pair"))))
+  :type '(alist
```
Comment (ninrod) [inline evil-surround.el]: I'm quite nervous about this change, mostly because I don't know sufficient elisp. 

Why did you change repeat to alist for starters?

Comment (wbolster) [inline evil-surround.el]: @ninrod have you actually tried comparing the generated ui in the customize interface before and after this change? :)

Comment (ninrod) [inline evil-surround.el]: no. I know that this pull is supposed to improve the customize interface so that the user is allowed to customize/add his own evil-surround-pairs-alist. I just want to understand how you are doing it. 

For starters, emacs does not tell me what `alist` is through `C-h f`. I find `repeat`, but no `alist`. 

Can you explain to me what exactly you are doing here and why it works? 

Thanks.

Comment (wbolster) [inline evil-surround.el]: see https://www.gnu.org/software/emacs/manual/html_node/elisp/Association-Lists.html

this `defcustom` uses dotted pairs, and these `:type` annotations tell the emacs customize interface how to build a ui for this configuration item; see screenshot.

please try it out before and after the change: `M-x customize-variable` then choose the variable `evil-surround-pairs-alist`.

Comment (ninrod) [inline evil-surround.el]: @wbolster thanks for the pointer, and I apologize in advance because I'm really an elisp beginner. I read the section but still, I did not find any `alist` function. 

Shouldn't the `alist` in the `(alist ...` you wrote be a function? If so, why isn't this function listed in the gnu doc section you've linked? I see all sorts of functions like `assoc`, `assq` but no `alist`. What am I missing here? Thanks

Comment (wbolster) [inline evil-surround.el]: https://www.gnu.org/software/emacs/manual/html_node/elisp/Composite-Types.html

Comment (ninrod) [inline evil-surround.el]: @wbolster thank you very much for the code.
